api/ccapi/include/dataset.h

```diff
@@ -87,23 +86,40 @@ class Dataset {
 };
 
 /**
- * @brief Factory creator with constructor for dataset
+ * @brief Create a Dataset object with given arguements
+ *
+ * @param type dataset type
+ * @param properties property representations
+ * @return std::unique_ptr<Dataset> created dataset
  */
 std::unique_ptr<Dataset>
 createDataset(DatasetType type,
               const std::vector<std::string> &properties = {});
 
 /**
- * @brief Factory creator with constructor for dataset
+ * @brief Create a Dataset object
+ *
+ * @param type dataset type
+ * @param path path to a file or folder
+ * @param properties property representations
+ * @return std::unique_ptr<Dataset> created dataset
  */
-std::unique_ptr<Dataset> createDataset(DatasetType type, const char *file);
+std::unique_ptr<Dataset>
+createDataset(DatasetType type, const char *path,
+              const std::vector<std::string> &properties = {});
```

Comment:
I wonder if the path can be properties.

Reply:
Path is in supported as props. This is to support constructing with char * argument. 
While this can be fully supported via path properties, here it is added for completeness (because callback producer must have to have it's property and later others might too).